ow_dynamic_terrain/src/TerrainModifier.cpp

```diff
@@ -243,29 +243,30 @@ bool TerrainModifier::applyImageToHeightmap(Heightmap* heightmap, const cv::Poin
     return false;
   }
 
-  auto heightmap_size = static_cast<int>(terrain->getSize());
-  auto left = max(center.x - image.cols / 2, 0);
-  auto top = max(center.y - image.rows / 2, 0);
-  auto right = min(center.x - image.cols / 2 + image.cols - 1, heightmap_size);
-  auto bottom = min(center.y - image.rows / 2 + image.rows - 1, heightmap_size);
+  int left = center.x - image.cols / 2;
+  int top = center.y - image.rows / 2;
+  int right = left + image.cols - 1;
+  int bottom = top + image.rows - 1;
 
   auto diff = OpenCV_Util::createZerosMatLike(image);
 
   bool change_occurred = false;
 
-  for (auto y = top; y <= bottom; ++y)
-    for (auto x = left; x <= right; ++x)
+  for (int y = top; y <= bottom; ++y)
+    for (int x = left; x <= right; ++x)
     {
-      auto pixel_value = image.at<float>(y - top, x - left);
+      // Using bottom - y instead of y - top because cv::Mat has a different
+      // layout than Ogre Heightmaps.
+      auto pixel_value = image.at<float>(bottom - y, x - left);
```

Comment:
The same calculation is used later for the diff image, so it might make sense to make them variables. 
```suggestion
      // Using bottom - y instead of y - top because cv::Mat has a different
      // layout than Ogre Heightmaps.
      auto iy = bottom - y;
      auto ix = x - left;
      auto pixel_value = image.at<float>(iy, ix);
```

Reply:
Good eye. Fixed.